awscli/customizations/gamelift/uploadbuild.py

```diff
@@ -102,6 +111,13 @@ def zip_directory(zipfile_name, source_root):
                     zf.write(full_path, relative_path)
 
 
+def validate_directory(source_root):
+    for path, dirs, files in os.walk(source_root):

```

Comment:
Is there a reason we can't just use `return os.path.isdir(source_root) and os.listdir(source_root)`?